UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarning/Admin/RadioNavigationalWarningAdminTest.cs

```diff
@@ -0,0 +1,158 @@
+            Assert.AreEqual(1, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithWarningTypeAndYearFilter_ThenReturnFilteredListAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+            IActionResult result = await _controller.Index(1, 2, 2024);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(1, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(1, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithValidPageNo_ThenReturnAsyncListBasedOnPageNo()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 3;
+            IActionResult result = await _controller.Index(2, null, null);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(3, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(3, adminListFilter.PageCount);
+            Assert.AreEqual(3, adminListFilter.SrNo);
+            Assert.AreEqual(2, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithInValidPageNo_ThenReturnEmptyListAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 3;
+            IActionResult result = await _controller.Index(4, null, null);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(0, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithValidAdminListRecordPerPage_ThenReturnValidRecordPerPageAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 5;
+            IActionResult result = await _controller.Index(1, null, null);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(5, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(2, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public void WhenCallIndexWithInValidAdminListRecordPerPage_ThenThrowDivideByZeroException()
```

Comment:
I don't think it's appropriate for the controller method to throw a divide by zero exception.  Can this be fixed?

Reply:
Thank you for adressing 

As this is the integration test, Here controller is not throwing the divide by zero exception, Exception is handled in-service class, but as this is the integration test we have to call the controller. 


And in the next sprint, we will develop a custom page for all exceptions.